codec-http/src/main/java/io/netty/handler/codec/http/cookie/ClientCookieEncoder.java

```diff
@@ -91,12 +127,64 @@ public String encode(Cookie... cookies) {
         }
 
         StringBuilder buf = stringBuilder();
-        for (Cookie c : cookies) {
-            if (c == null) {
-                break;
+        if (strict) {
+            List<Cookie> cookiesList = new ArrayList<Cookie>(cookies.length);
+            for (Cookie c : cookies) {
+                if (c == null) {
+                    break;
+                }
+
+                cookiesList.add(c);
+            }
+            Collections.sort(cookiesList, COOKIE_COMPARATOR);
+            for (Cookie c : cookiesList) {
```

Comment:
If we iterator over an array, we could save an Iterator here and use a good old for (int i = 0; ...)